client/src/main/java/io/pravega/client/stream/impl/EventStreamReaderImpl.java

```diff
@@ -102,7 +103,8 @@
                     try {
                         buffer = segmentReader.read(waitTime);
                     } catch (EndOfSegmentException e) {
-                        handleEndOfSegment(segmentReader);
+                        boolean fetchSuccessors = e.getErrorType().equals(ErrorType.END_OF_SEGMENT_REACHED);
```

Comment:
If an EndOfSegment is encountered because it read the end of the segment, but that was still one of the segments in the end streamcut then it should still not fetch the successors. So handleEndOfSegment needs to be able to work out if the segment is a terminal one anyway. So adding this boolean doesn't actually provide more information.

Reply:
Based on the discussions, an EndOfSegment from SSS cannot be obtained before the `EndStreamCut` since custom StreamCut objects cannot be created by the user.
Also, the code ensures successors are not fetched if `endStreamCut` points to `EndOfSegment`